tests/unit/test_index_contents_parse.py

```diff
@@ -0,0 +1,426 @@
+                ),
+            ),
+            id="single item content before",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})
+# Other content
+- [other title 1](other value 1)
+""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+            ),
+            id="single item content after",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})
+# Contents
+- [other title 1](other value 1)
+""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+            ),
+            id="single item content after with duplicate heading",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})
+- [{(title_2 := 'title 2')}]({(value_2 := 'value 2')})
+""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_2, reference_value=value_2, rank=1
+                ),
+            ),
+            id="multiple items flat",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})
+  - [{(title_2 := 'title 2')}]({(value_2 := 'value 2')})
+""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=2, reference_title=title_2, reference_value=value_2, rank=1
+                ),
+            ),
+            id="multiple items nested",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})
+ - [{(title_2 := 'title 2')}]({(value_2 := 'value 2')})
+""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=1, reference_title=title_2, reference_value=value_2, rank=1
+                ),
+            ),
+            id="multiple items nested alternate spacing single space",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})
+- [{(title_2 := 'title 2')}]({(value_2 := 'value 2')})
+""",
```

Comment:
Shouldn't it be:
f"""# Contents
- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})

- [{(title_2 := 'title 2')}]({(value_2 := 'value 2')})
""" ?

Reply:
good catch, though there shouldn't be a line between contents and the first line since that is what the earlier test covers. Should be fixed on next push

Reply:
I've updated my comment, I forgot to put it in a code block. Thx